driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -111,8 +142,31 @@ public BsonDocument getResumeToken() {
      *
      * @return the namespace
      */
+    @BsonIgnore @Nullable
     public MongoNamespace getNamespace() {
-        return namespace;
+        return namespaceDocument != null
```

Comment:
Throw IllegalStateException here if either db or coll are not in namespaceDocument.  Otherwise it will throw a NullPointerException calling getValue, which is harder to understand.

Reply:
Done